webkit2/tests/tests.lisp

```diff
@@ -0,0 +1,258 @@
+  (with-js-transform-result (%result% "true")
+    (is (eq :true %result%)))
+  (setf webkit::*js-true-value* t))
+
+;;; Numbers
+
+(def-test integers (:suite js-tests)
+  (with-js-transform-result (%result% "0")
+    (is (= 0 %result%)))
+  (with-js-transform-result (%result% "8")
+    (is (= 8 %result%)))
+  (with-js-transform-result (%result% "-6")
+    (is (= -6 %result%)))
+  (with-js-transform-result (%result% "Math.pow(10, 100)")
+    (is (= 1.0000000000000002d100 %result%)))
+  (with-js-transform-result (%result% "-Math.pow(10, 100)")
+    (is (= -1.0000000000000002d100 %result%))))
+
+;; REVIEW: CCL? ECL?
+#+sbcl
+(def-test special-numbers (:suite js-tests)
+  (with-js-transform-result (%result% "NaN")
+    (is (sb-ext:float-nan-p %result%)))
+  (with-js-transform-result (%result% "Infinity")
+    (is (sb-ext:float-infinity-p %result%))
+    (is (equal 1.0d0 (float-sign %result% 1.0))))
+  (with-js-transform-result (%result% "-Infinity")
+    (is (sb-ext:float-infinity-p %result%))
+    (is (equal -1.0d0 (float-sign %result% 1.0)))))
+
+(def-test fractional-number (:suite js-tests)
+  (with-js-transform-result (%result% "5.3")
+    (is (= 5.3d0 %result%))))
+
+(def-test periodic-number (:suite js-tests)
+  (with-js-transform-result (%result% "var num = 5/3; num")
+    (is (equal 1.6666666666666667d0 %result%))))
+
+;;; Strings
+
+(def-test simple-string (:suite js-tests)
+  (with-js-transform-result (%result% "\"hello\"")
+    (is (equal "hello" %result%))))
+
+(def-test escaped-string (:suite js-tests)
+  (with-js-transform-result (%result% "\"hello\\nthere\"")
+    (is (equal "hello
+there" %result%))))
+
+(def-test templated-string (:suite js-tests)
+  (with-js-transform-result (%result% "var num = 5; `${num} + ${num} = ${10}`")
+    (is (equal "5 + 5 = 10" %result%))))
+
+(def-test concatenated-string (:suite js-tests)
+  (with-js-transform-result (%result% "\"hello \" + \"there!\"")
+    (is (equal "hello there!" %result%))))
+
+;;; Arrays
+
+(def-test simple-array (:suite js-tests)
+  (with-js-transform-result (%result% "[1, 2, 3, 4, 5]")
+    (is (equal (list 1 2 3 4 5) %result%)))
+  (with-js-transform-result (%result% "[\"h\", \"e\", \"l\", \"l\", \"o\"]")
+    (is (equal (list "h" "e" "l" "l" "o") %result%)))
+  (with-js-transform-result (%result% "[true, false, true, true, false]")
+    (is (equal (list t nil t t nil) %result%))))
+
+(def-test everything-array (:suite js-tests)
+  (with-js-transform-result (%result% "[true, false, undefined, null, 100000, \"hello\", {one: 1}]")
+    (is (equal (list t nil :undefined :null 100000 "hello" '(("one" . 1))) %result%))))
+
+(def-test everything-array-vector (:suite js-tests)
+  (setf webkit:*js-array-type* :vector)
+  (with-js-transform-result (%result% "[true, false, undefined, null, 100000, \"hello\", {one: 1}]")
+    (is (equalp (vector t nil :undefined :null 100000 "hello" '(("one" . 1))) %result%)))
+  (setf webkit:*js-array-type* :list))
```

Comment:
Use unwind-protect to ensure the global is reset on error.

Ideally, we should figure out the dynamic binding properly.